middleware/authentication.go

```diff
@@ -72,15 +72,17 @@ func MakeHTTPBasicAuthenticationMapMW(credentials map[string]string, logger log.
 // If there is no such header, the request is not allowed.
 // If the password is correct, the username is added into the context:
 //   - username: username extracted from the token
-func MakeHTTPBasicAuthenticationMW(passwordToMatch string, logger log.Logger) func(http.Handler) http.Handler {
+func MakeHTTPBasicAuthenticationMW(passwordToMatch []string, logger log.Logger) func(http.Handler) http.Handler {
```

Comment:
Different passwords should be used by different users (with different usernames). You should use MakeHTTPBasicAuthenticationMapMW (define right above this one)